ocs_ci/ocs/ui/helpers_ui.py

```diff
@@ -95,6 +100,101 @@ def format_locator(locator, string_to_insert):
+        logger.info("Storage Class with Encryption")
+        base_ui_obj.do_click(pvc_loc["encryption"])
+        logger.info("Checking if 'Change connection details' option is available")
+        conn_details = base_ui_obj.check_element_text(
+            expected_text="Change connection details"
+        )
+        if conn_details:
+            logger.info("Click on Change Connection Details")
+            base_ui_obj.do_click(pvc_loc["connections-details"])
+        logger.info("Storage Class Service Name")
+        base_ui_obj.do_clear(pvc_loc["service-name"])
+        base_ui_obj.do_send_keys(pvc_loc["service-name"], "vault")
+        logger.info("Storage Class Address")
+        base_ui_obj.do_clear(pvc_loc["kms-address"])
+        base_ui_obj.do_send_keys(pvc_loc["kms-address"], "https://vault.qe.rh-ocs.com/")
+        logger.info("Storage Class Port")
+        base_ui_obj.do_clear(pvc_loc["kms-port"])
+        base_ui_obj.do_send_keys(pvc_loc["kms-port"], "8200")
+        logger.info("Click on Advanced Settings")
+        base_ui_obj.do_click(pvc_loc["advanced-settings"])
+        logger.info("Enter Backend Path")
+        base_ui_obj.do_clear(pvc_loc["backend-path"])
+        base_ui_obj.do_send_keys(pvc_loc["backend-path"], backend_path)
+        logger.info("Enter TLS Server Name")
+        base_ui_obj.do_clear(pvc_loc["tls-server-name"])
+        base_ui_obj.do_send_keys(pvc_loc["tls-server-name"], "vault.qe.rh-ocs.com")
+        logger.info("Enter Vault Enterprise Namespace")
+        base_ui_obj.do_clear(pvc_loc["vault-enterprise-namespace"])
+        base_ui_obj.do_send_keys(pvc_loc["vault-enterprise-namespace"], namespace)
+        logger.info("Selecting CA Certificate")
+        base_ui_obj.do_click(pvc_loc["browse-ca-certificate"])
+        time.sleep(1)
+        write(os.path.abspath(constants.VAULT_CA_CERT_PEM))
+        time.sleep(1)
+        press("enter")
+        time.sleep(1)
+        logger.info("Selecting Client Certificate")
+        base_ui_obj.do_click(pvc_loc["browse-client-certificate"])
+        time.sleep(1)
+        write(os.path.abspath(constants.VAULT_CLIENT_CERT_PEM))
+        time.sleep(1)
+        press("enter")
+        time.sleep(1)
+        logger.info("Selecting Client Private Key")
+        base_ui_obj.do_click(pvc_loc["browse-client-private-key"])
+        time.sleep(1)
+        write(os.path.abspath(constants.VAULT_PRIVKEY_PEM))
+        time.sleep(1)
+        press("enter")
+        time.sleep(1)
+        logger.info("Saving Key Management Service Advanced Settings")
+        base_ui_obj.do_click(pvc_loc["save-advanced-settings"], enable_screenshot=True)
+        time.sleep(1)
+        logger.info("Save Key Management Service details")
+        base_ui_obj.do_click(pvc_loc["save-service-details"], enable_screenshot=True)
+        time.sleep(1)
+    logger.info("Creating Storage Class with Encryption")
+    base_ui_obj.do_click(pvc_loc["create"])
+    time.sleep(1)
+
+    return sc_name
```

Comment:
You can add before return the verify of storageclass inside this function and return the name and the status together. This way each time you create the storageclass it is already verified that it is created.

Reply:
done